contrib/drivers/ceph/replication.go

```diff
@@ -0,0 +1,407 @@
+
+	err = cephpeeraddsession.Run("rbd mirror pool peer add rbd client.remote@remote")
+	if err != nil {
+	}
+
+	cephpeeraddsession.Close()
+
+	backuppeersession, err := backupclient.NewSession()
+
+	if err := backuppeersession.Run("rbd --cluster remote mirror pool peer add rbd client.ceph@ceph"); err != nil {
+	}
+	backuppeersession.Close()
+	PrimaryVolumeId := opensdsPrefix + opt.PrimaryVolumeId
+	SecondaryVolumeId := opensdsPrefix + opt.SecondaryVolumeId
+	PoolId := opt.PoolId
+	AvailbilityZone := opt.AvailabilityZone
+	profileid := opt.ProfileId
+
+	additionalCephData := map[string]string{
+		"PrimaryIP": r.conf.IPaddresshost,
+	}
+
+	additionalBackupData := map[string]string{
+		"RemoteIP": r.conf.Ipaddresspeer,
+	}
+
+	return &model.ReplicationSpec{
+		PrimaryVolumeId:                PrimaryVolumeId,
+		SecondaryVolumeId:              SecondaryVolumeId,
+		PoolId:                         PoolId,
+		ProfileId:                      profileid,
+		AvailabilityZone:               AvailbilityZone,
+		PrimaryReplicationDriverData:   additionalCephData,
+		SecondaryReplicationDriverData: additionalBackupData,
+	}, nil
+
+}
+
+// Delete replication
+func (r *ReplicationDriver) DeleteReplication(opt *pb.DeleteReplicationOpts) error {
+
+	cephconfig := &ssh.ClientConfig{
+		User: r.conf.Username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(r.conf.Password),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	cephclient, err := ssh.Dial("tcp", r.conf.HostdailIP, cephconfig)
+	if err != nil {
+		log.Error("Failed to dial: " + err.Error())
+
+	}
+	backupconfig := &ssh.ClientConfig{
+		User: r.conf.Username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(r.conf.Password),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	backupclient, error := ssh.Dial("tcp", r.conf.PeerdailIP, backupconfig)
+
+	if error != nil {
+		log.Error("Failed to dial: " + error.Error())
+
+	}
+
+	cephsession, err := cephclient.NewSession()
+
+	if err := cephsession.Run("systemctl stop ceph-rbd-mirror@ceph"); err != nil {
+		logs.Error(err)
+	}
+	cephsession.Close()
+
+	backupsession, err := backupclient.NewSession()
+
+	if err := backupsession.Run("systemctl stop ceph-rbd-mirror@remote"); err != nil {
+		logs.Error(err)
+	}
+
+	backupsession.Close()
+
+	return nil
+}
+
+// Start Replication
+func (r *ReplicationDriver) EnableReplication(opt *pb.EnableReplicationOpts) error {
+
+	cephconfig := &ssh.ClientConfig{
+		User: r.conf.Username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(r.conf.Password),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	cephclient, err := ssh.Dial("tcp", r.conf.HostdailIP, cephconfig)
+	if err != nil {
+		log.Error("Failed to dial: " + err.Error())
+
+	}
+
+	cephsession, err := cephclient.NewSession()
+	if err != nil {
+		log.Error("Failed to create session: " + err.Error())
+
+	}
+
+	volumename := opensdsPrefix + opt.PrimaryVolumeId
+
+	cmd := "rbd mirror image enable rbd/" + volumename + " --pool rbd"
```

Comment:
Is the command 'rbd' always in PATH? Can it be in non-default location as per user installation of ceph